frontend/src/metabase/dashboard/components/RefreshWidget/RefreshWidget.tsx

```diff
@@ -0,0 +1,79 @@
+import { useState } from "react";
+import { usePrevious, useUpdateEffect } from "react-use";
+import { t } from "ttag";
+
+import { Box, Popover } from "metabase/ui";
+
+import { RefreshOption } from "./RefreshOption";
+import RefreshWidgetS from "./RefreshWidget.module.css";
+import { RefreshWidgetTarget } from "./RefreshWidgetTarget";
+
+const toSeconds = (minutes: number) => minutes * 60;
+
+const OPTIONS = [
+  { name: t`Off`, period: null },
+  { name: t`1 minute`, period: toSeconds(1) },
+  { name: t`5 minutes`, period: toSeconds(5) },
+  { name: t`10 minutes`, period: toSeconds(10) },
+  { name: t`15 minutes`, period: toSeconds(15) },
+  { name: t`30 minutes`, period: toSeconds(30) },
+  { name: t`60 minutes`, period: toSeconds(60) },
+];
+
+export const RefreshWidget = ({
+  setRefreshElapsedHook,
+  period,
+  onChangePeriod,
+}: {
+  setRefreshElapsedHook?: (hook: (elapsed: number | null) => void) => void;
+  period: number | null;
+  onChangePeriod: (period: number | null) => void;
```

Comment:
nit: wondering if we have a naming convention on this? (e.g. the props in the above diff is `onRefreshPeriodChange`, so this could be `onPeriodChange` too or vice-versa)

Reply:
One thing that I'm doing with #42777 will be to standardize this. Don't worry, it's definitely a thought I've had as well 😄